packages/ai-chat/src/common/chat-model.ts

```diff
@@ -954,13 +988,17 @@ class ChatResponseImpl implements ChatResponse {
                 );
                 return undefined;
             })
-            .filter(text => text !== undefined)
+            .filter(text => (text !== undefined || text === ''))
```

Comment:
This change does nothing as `''` is already `!== undefined`, you probably wanted to do this to filter empty string?

```suggestion
            .filter(text => (text !== undefined && text !== ''))
```

At the moment we have a lot of whitespace where the tool calls were in the past:

```
 "query": "I'll help you check the contents of your workspace using the getWorkspaceFileList function for the root directory.
                
                
                
                
                
                Here are the files and directories in your workspace root:
                
                Files:
                - CHANGELOG.md
                ...
```

With the suggested change this changes to:


```
      "query": "I'll help you check the contents of your workspace using the getWorkspaceFileList function for the root directory.
                
                
                
                Here are the main files and directories in your workspace root:
```

Which is still a lot but at least it's less